hyunjinjeon/app.js

```diff
@@ -0,0 +1,153 @@
+const cors = require("cors");
+const morgan =  require("morgan");
+const dotenv = require("dotenv");
+const { DataSource } = require('typeorm'); 
+
+//Custom
+app = express();
+app.use(express.json());
+app.use(cors());
+app.use(morgan('combined'));
+dotenv.config()
+
+const appDataSource = new DataSource({
+    type: process.env.TYPEORM_CONNECTION,
+    host: process.env.TYPEORM_HOST,
+    port: process.env.TYPEORM_PORT,
+    username: process.env.TYPEORM_USERNAME,
+    password: process.env.TYPEORM_PASSWORD,
+    database: process.env.TYPEORM_DATABASE
+})
+
+//Error handling
+appDataSource.initialize()
+  .then(()=>{
+    console.log("Data Source has been initialized!")
+  });
+
+
+// 유저 회원 가입 엔드포인트
+app.post('/signup',async(req,res)=>{
+    const {name,email,password,age} = req.body
+    await appDataSource.query(
+        `INSERT INTO users (
+          user_name,
+          user_email,
+          user_password,
+          user_age
+       ) VALUES(?,?,?,?); 
+        `,[name,email,password,age]);
+
+    res.status(201).json({message:"userCreated"});
+})
+
+
+// 게시물등록 엔드포인트
+app.post('/posts',async(req,res)=>{
+    const {title,content} = req.body
+    await appDataSource.query(
+      `INSERT INTO posts (
+        title,
+        content
+     ) VALUES(?,?);
+    `,[title,content]);
+    res.status(201).json({message:"postCreated"});
+ })
+
+
+//게시물 전체조회 엔드포인트
+ app.get('/posts',async(req,res)=>{
+  await appDataSource.query(
+    `SELECT
+        users.user_name,
+        posts.title,
+        posts.created_at
+     FROM posts
+     LEFT JOIN users ON users.user_id = posts.user_id;
+      `,(err,rows)=>
+      {res.status(200).json(rows);});
+  });
+
+
+
+//특정유저 게시물 조회 엔드포인트
+app.get("/userpost/:userId", async (req,res) => {
+  const userId = req.params.userId;
+    await appDataSource.query(
+      `SELECT 
+        users.id as userId,
+        users.profile_image as userporfileImage,
+        JSON_ARRAYAGG( 
+        JSON_OBJECT(    
+          'postingId',posts.id,
+          'postingTitle',posts.title,
+          'postingContent',posts.content)) as posting
+      FROM users
+      LEFT JOIN posts ON users.id = posts.user_id 
+      where users.id=${userId}
+      GROUP BY users.id;
+      `,(err,row)=>
+      {res.status(200).json({data:row});});
+      })
+
+
+//게시물 수정 엔드포인트
+app.patch("/modifypost/:postId", async (req, res) => {
+  const postId = req.params.postId;
+  const {content} = req.body;
+  await appDataSource.query(
+    `UPDATE posts SET
+      content = ?
+    WHERE id = ${postId}
+    `,[content]);
+  const post = await appDataSource.query(
+    `SELECT
+      users.id as userId,
+      users.name as userName,
+      posts.id as postingId,
+      posts.title as postingTitle,
+      posts.content as postingContent
+    FROM posts INNER JOIN users on users.id = posts.user_id where posts.id like ${postId};`,
+    );
+    const result = post[0];
+    res.status(204).json({data : result});
+  })
```

Comment:
형진님! 저희가 보내는 patch, 혹은 delete에 어떤 작업을 의미하는지 이미 내포되어 있기 때문에 
엔드포인트에 수정한다 라는 것을 명시해주지 않아도 됩니다. 오히려 의미의 중복이 발생할 수 있기 때문에
해당 부분은 지양해야 합니다 !
```suggestion
app.patch("/posts/:postId", async (req, res) => {
  const postId = req.params.postId;
  const {content} = req.body;
  await appDataSource.query(
    `UPDATE posts SET
      content = ?
    WHERE id = ${postId}
    `,[content]);
  const post = await appDataSource.query(
    `SELECT
      users.id as userId,
      users.name as userName,
      posts.id as postingId,
      posts.title as postingTitle,
      posts.content as postingContent
    FROM posts INNER JOIN users on users.id = posts.user_id where posts.id like ${postId};`,
    );
    const result = post[0];
    res.status(204).json({data : result});
  })
```